presto-hive/src/test/java/com/facebook/presto/hive/TestMergeJoinPlan.java

```diff
@@ -0,0 +1,210 @@
+            queryRunner.execute("DROP TABLE IF EXISTS test_join_order2");
+        }
+    }
+
+    @Test
+    public void testDifferentSortByKey()
+    {
+        QueryRunner queryRunner = getQueryRunner();
+
+        try {
+            queryRunner.execute("CREATE TABLE test_join_customer3 WITH ( \n" +
+                    "  bucket_count = 4, bucketed_by = ARRAY['custkey'], \n" +
+                    "  sorted_by = ARRAY['name'], partitioned_by=array['ds']) AS \n" +
+                    "SELECT *, '2021-07-11' as ds FROM tpch.sf1.customer LIMIT 1000");
+
+            queryRunner.execute("CREATE TABLE test_join_order3 WITH ( \n" +
+                    "  bucket_count = 4, bucketed_by = ARRAY['custkey'], \n" +
+                    "  sorted_by = ARRAY['custkey'], partitioned_by=array['ds']) AS \n" +
+                    "SELECT *, '2021-07-11' as ds FROM tpch.sf1.\"orders\" LIMIT 1000");
+
+            // merge join can't be enabled
+            assertPlan(
+                    mergeJoinEnabled(),
+                    "select * from test_join_customer3 join test_join_order3 on test_join_customer3.custkey = test_join_order3.custkey",
+                    joinPlan("test_join_customer3", "test_join_order3", ImmutableList.of("custkey"), ImmutableList.of("custkey"), false));
+        }
+        finally {
+            queryRunner.execute("DROP TABLE IF EXISTS test_join_customer3");
+            queryRunner.execute("DROP TABLE IF EXISTS test_join_order3");
+        }
+    }
+
+    @Test
+    public void testMultipleSortByKeys()
+    {
+        QueryRunner queryRunner = getQueryRunner();
+
+        try {
+            queryRunner.execute("CREATE TABLE test_join_customer4 WITH ( \n" +
+                    "  bucket_count = 4, bucketed_by = ARRAY['custkey'], \n" +
+                    "  sorted_by = ARRAY['custkey', 'name'], partitioned_by=array['ds']) AS \n" +
+                    "SELECT *, '2021-07-11' as ds FROM tpch.sf1.customer LIMIT 1000");
+
+            queryRunner.execute("CREATE TABLE test_join_order4 WITH ( \n" +
+                    "  bucket_count = 4, bucketed_by = ARRAY['custkey'], \n" +
+                    "  sorted_by = ARRAY['custkey'], partitioned_by=array['ds']) AS \n" +
+                    "SELECT *, '2021-07-11' as ds FROM tpch.sf1.\"orders\" LIMIT 1000");
+
+            // merge join can be enabled
+            assertPlan(
+                    mergeJoinEnabled(),
+                    "select * from test_join_customer4 join test_join_order4 on test_join_customer4.custkey = test_join_order4.custkey",
+                    joinPlan("test_join_customer4", "test_join_order4", ImmutableList.of("custkey"), ImmutableList.of("custkey"), true));
+        }
+        finally {
+            queryRunner.execute("DROP TABLE IF EXISTS test_join_customer4");
+            queryRunner.execute("DROP TABLE IF EXISTS test_join_order4");
+        }
+    }
+
+    // todo add test for multiple join keys
```

Comment:
can you add this test?

Reply:
got it. Can you add a test for whatever the expected result is for multiple join keys, and then we can update once support is added?  Just want to make sure all behaviors are well defined.